image/preseed/preseed_linux.go

```diff
@@ -275,6 +279,29 @@ func prepareCore20Mountpoints(prepareImageDir, tmpPreseedChrootDir, snapdSnapBlo
 		{"--bind", writable, underPreseed("writable")},
 	}
 
+	if sysfsOverlay != "" {
+		// bind mount only permitted directories under sys/class and sys/devices
+		for _, dir := range []string{
+			"sys/class/backlight", "sys/class/bluetooth", "sys/class/gpio",
+			"sys/class/leds", "sys/class/ptp", "sys/class/pwm",
+			"sys/class/rtc", "sys/class/video4linux", "sys/devices/platform",
+			"sys/devices/pci0000:00"} {
+			info, err := os.Stat(underOverlay(dir))
+			if err == nil && info.IsDir() {
+				// ensure dir exists
+				if _, err := os.Stat(underPreseed(dir)); errors.Is(err, os.ErrNotExist) {
+					err := os.MkdirAll(underPreseed(dir), os.ModePerm)
```

Comment:
Do you really need the `os.Stat()` call? `os.MkdirAll` will just do nothing if the directory already exists.

(sorry for not pointing it out earlier, but I also learnt this only recently :-) )

Reply:
Nice one, `os.Stat()` is not removed